vllm/v1/executor/abstract.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC, abstractmethod
+from typing import Tuple
+
+from vllm.config import VllmConfig
+from vllm.v1.outputs import ModelRunnerOutput
+
+
+class Executor(ABC):
```

Comment:
at a high level, i think `class Executor` should only have:

```python
class Executor:
    def __init__(self, vllm_config: VllmConfig):

    def collective_rpc(self, ...):
```

the engine should only interact with the executor via `collective_rpc`

Reply:
Leaving this for now, until we axe `UniprocExecutor`

Reply:
sounds good to me right now. please keep in mind and refactor later.